fl4health/clients/fed_prox_client.py

```diff
@@ -93,12 +80,15 @@ def set_parameters(self, parameters: NDArrays, config: Config, fitting_round: bo
 
         super().set_parameters(server_model_state, config, fitting_round)
 
+    def update_before_train(self, current_server_round: int) -> None:
```

Comment:
Moving the setting of the `initial_tensors` to the `update_before_train` function so that it isn't done during validation. It wasn't causing any issues, but it's unnecessary and this function conveniently isolates both the `set_parameters` functionality and  ensures that the tensors are saved only once before local training starts